nft/ethereum_payment_obligation.go

```diff
@@ -80,45 +88,155 @@ func newEthereumPaymentObligation(
 	}
 }
 
-func (s *ethereumPaymentObligation) prepareMintRequest(ctx context.Context, tokenID TokenID, documentID []byte, depositAddress string, proofFields []string) (MintRequest, error) {
-	model, err := s.docSrv.GetCurrentVersion(ctx, documentID)
+func (s *ethereumPaymentObligation) prepareMintRequest(ctx context.Context, tokenID TokenID, cid identity.CentID, req MintNFTRequest) (mreq MintRequest, err error) {
+	model, err := s.docSrv.GetCurrentVersion(ctx, req.DocumentID)
 	if err != nil {
-		return MintRequest{}, err
+		return mreq, err
 	}
 
 	corDoc, err := model.PackCoreDocument()
 	if err != nil {
-		return MintRequest{}, err
+		return mreq, err
 	}
 
-	proofs, err := s.docSrv.CreateProofs(ctx, documentID, proofFields)
+	docProofs, err := s.docSrv.CreateProofs(ctx, req.DocumentID, req.ProofFields)
 	if err != nil {
-		return MintRequest{}, err
+		return mreq, err
+	}
+
+	nftProofs, err := generateNFTProofs(corDoc, req, tokenID, cid)
+	if err != nil {
+		return mreq, err
+	}
+
+	var pfs []*proofspb.Proof
+	pfs = append(pfs, docProofs.FieldProofs...)
+	for _, p := range nftProofs {
+		p := p
+		pfs = append(pfs, &p)
 	}
 
-	toAddress := common.HexToAddress(depositAddress)
+	toAddress := common.HexToAddress(req.DepositAddress)
 
 	anchorID, err := anchors.ToAnchorID(corDoc.CurrentVersion)
 	if err != nil {
-		return MintRequest{}, err
+		return mreq, err
 	}
 
 	rootHash, err := anchors.ToDocumentRoot(corDoc.DocumentRoot)
 	if err != nil {
-		return MintRequest{}, err
+		return mreq, err
 	}
 
-	requestData, err := NewMintRequest(tokenID, toAddress, anchorID, proofs.FieldProofs, rootHash)
+	requestData, err := NewMintRequest(tokenID, toAddress, anchorID, pfs, rootHash)
 	if err != nil {
-		return MintRequest{}, err
+		return mreq, err
 	}
 
 	return requestData, nil
 
 }
 
+// generateNFTProofs generates required NFT uniqueness proofs
```

Comment:
This method doesn't just do the uniqueness.

```suggestion
// generateNFTProofs generates required NFT proofs
```